equipment-models/src/main/java/com/telecominfraproject/wlan/equipment/models/ApElementConfiguration.java

```diff
@@ -56,18 +56,13 @@ public static ApElementConfiguration createWithDefaults(String elementVersion) {
 	private ApElementConfiguration() {
 		super(EquipmentType.AP);
 
-		/* We populate the radio map */
+		/* Populate the radio maps for all valid radio types */
 		radioMap = new EnumMap<>(RadioType.class);
-		radioMap.put(RadioType.is5GHzL, ElementRadioConfiguration.createWithDefaults(RadioType.is5GHzL));
-		radioMap.put(RadioType.is5GHzU, ElementRadioConfiguration.createWithDefaults(RadioType.is5GHzU));
-		radioMap.put(RadioType.is2dot4GHz, ElementRadioConfiguration.createWithDefaults(RadioType.is2dot4GHz));
-
-		/* We populate the advanced radio map */
 		advancedRadioMap = new EnumMap<>(RadioType.class);
-		advancedRadioMap.put(RadioType.is2dot4GHz, RadioConfiguration.createWithDefaults(RadioType.is2dot4GHz));
-		advancedRadioMap.put(RadioType.is5GHzL, RadioConfiguration.createWithDefaults(RadioType.is5GHzL));
-		advancedRadioMap.put(RadioType.is5GHzU, RadioConfiguration.createWithDefaults(RadioType.is5GHzU));
-
+		for(RadioType radioType: RadioType.validValues()) {
+			radioMap.put(radioType, ElementRadioConfiguration.createWithDefaults(radioType));
+			advancedRadioMap.put(radioType, RadioConfiguration.createWithDefaults(radioType));
+		}
```

Comment:
Nice try :) but it will create a 4th element- for is5GHz, which I was trying to avoid here.